app/services/category_hashtag_data_source.rb

```diff
@@ -71,17 +71,14 @@ def self.search_sort(search_results, term)
   end
 
   def self.search_without_term(guardian, limit)
-    Category
-      .includes(:parent_category)
-      .secured(guardian)
-      .joins(
-        "LEFT JOIN category_users ON category_users.user_id = #{guardian.user.id}
-        AND category_users.category_id = categories.id",
-      )
-      .where(
-        "category_users.notification_level IS NULL OR category_users.notification_level != ?",
-        CategoryUser.notification_levels[:muted],
-      )
+    muted_category_ids = CategoryUser.muted_category_ids(guardian.user)
+    category_query = Category.includes(:parent_category).secured(guardian)
+
+    if muted_category_ids.any?
+      category_query = category_query.where("categories.id NOT IN (?)", muted_category_ids)
```

Comment:
I don't think we need to pluck the ids out of the database for this use case. Instead, we can just use the muted category ids SQL as a subplan. 

You can see how I did it in 

https://github.com/discourse/discourse/blob/ae2153b330a5d8bef41fa202a221b43c3b826166/app/models/topic.rb#L645

Reply:
Ah right, thanks will change

Reply:
Done!